SokolTextGame/Objects.cs

```diff
@@ -8,28 +7,54 @@ public interface IObject
         public string Description { get; }
         public string[] LinesObjectSays { get; }
         public string[] ItemsForSale { get; }
+        public bool ObjectIsAlive { get; set; }
+        public int HealtObject { get; set; }
+        public int DamageObject { get; set; }
     }
 }
 public class Guard : IObject
 {
     public string Name => "guard";
     public string Description => "The guard is a man in steel armor and with a huge two-handed club.\n" +
-        "You don't want to fight him, but you can ask him something.\n";
+    "You don't want to fight him, but you can ask him something.\n";
     public string[] LinesObjectSays => new[] { "What is that smell? Isn't that shit on your shoe?\n",
     "Life is a nightmare that prevents one from sleeping.\n",
     "Be yourself; everyone else is already taken\n",
     "It is awfully hard work doing nothing.\n",
     "Watermelon – it’s a good fruit. You eat, you drink, you wash your face.\n",
     "Death is hereditary.\n"};
     public string[] ItemsForSale { get; } = null;
+    public bool ObjectIsAlive { get; set; } = true;
+    public int HealtObject { get; set; } = 100;
+    public int DamageObject { get; set; } = 50;
 }
 
 public class Shopkeeper : IObject
 {
     public string Name => "shopkeeper";
-    public string Description => "A grandfather, with a big gray beard and strong facial features.\n" +
-        "He had met many travelers, seen many battles, and knew many secrets about weapons,\n" +
-        "but he was not very willing to share those secrets.\n";
-    public string[] LinesObjectSays => new[] {"Stop talking and buy something already!"};
+    public string Description => "A old man, with a big gray beard and strong facial features.\n" +
+    "He had met many travelers, seen many battles, and knew many secrets about weapons,\n" +
+    "but he was not very willing to share those secrets.\n";
+    public string[] LinesObjectSays => new[] { "Stop talking and buy something already!" };
     public string[] ItemsForSale { get; } = new[] { "axe", "sword", "staff" };
+    public bool ObjectIsAlive { get; set; } = true;
+
+    public int HealtObject { get; set; } = 100;
+    public int DamageObject { get; set; } = 50;
```

Comment:
You have two options here.
1. Make the damage for the NPCs the player is not supposed to kill outrageous, like 5000 or something.
2. Make sure at least the buy command checks the dead-or-alive status.

As it is right now, you need to either make it impossible to kill NPCs, or check the dead-or-alive status on all the commands. As of right now, it's possible to buy something from an object even if they're dead.